qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/MSDTemplateBrowser.cpp

```diff
@@ -0,0 +1,347 @@
+  m_fitType->addSubProperty(m_yiMsd);
+  m_fitType->addSubProperty(m_yiSigma);
+  ScopedFalse _false(m_emitEnumChange);
+  m_enumManager->setValue(m_fitType, 3);
+}
+
+void MSDTemplateBrowser::removeYi() {
+  m_fitType->removeSubProperty(m_yiHeight);
+  m_fitType->removeSubProperty(m_yiMsd);
+  m_fitType->removeSubProperty(m_yiSigma);
+  ScopedFalse _false(m_emitEnumChange);
+  m_enumManager->setValue(m_fitType, 0);
+}
+
+int MSDTemplateBrowser::getCurrentDataset() {
+  return m_presenter.getCurrentDataset();
+}
+
+void MSDTemplateBrowser::setGaussianHeight(double value, double error) {
+  setParameterPropertyValue(m_gaussianHeight, value, error);
+}
+
+void MSDTemplateBrowser::setGaussianMsd(double value, double error) {
+  setParameterPropertyValue(m_gaussianMsd, value, error);
+}
+
+void MSDTemplateBrowser::setPetersHeight(double value, double error) {
+  setParameterPropertyValue(m_petersHeight, value, error);
+}
+
+void MSDTemplateBrowser::setPetersMsd(double value, double error) {
+  setParameterPropertyValue(m_petersMsd, value, error);
+}
+
+void MSDTemplateBrowser::setPetersBeta(double value, double error) {
+  setParameterPropertyValue(m_petersBeta, value, error);
+}
+
+void MSDTemplateBrowser::setYiHeight(double value, double error) {
+  setParameterPropertyValue(m_yiHeight, value, error);
+}
+
+void MSDTemplateBrowser::setYiMsd(double value, double error) {
+  setParameterPropertyValue(m_yiMsd, value, error);
+}
+
+void MSDTemplateBrowser::setYiSigma(double value, double error) {
+  setParameterPropertyValue(m_yiSigma, value, error);
+}
+
+void MSDTemplateBrowser::setFunction(const QString &funStr) {
+  m_presenter.setFunction(funStr);
+}
+
+IFunction_sptr MSDTemplateBrowser::getGlobalFunction() const {
+  return m_presenter.getGlobalFunction();
+}
+
+IFunction_sptr MSDTemplateBrowser::getFunction() const {
+  return m_presenter.getFunction();
+}
+
+void MSDTemplateBrowser::setNumberOfDatasets(int n) {
+  m_presenter.setNumberOfDatasets(n);
+}
+
+int MSDTemplateBrowser::getNumberOfDatasets() const {
+  return m_presenter.getNumberOfDatasets();
+}
+
+void MSDTemplateBrowser::setDatasetNames(const QStringList &names) {
+  m_presenter.setDatasetNames(names);
+}
+
+QStringList MSDTemplateBrowser::getGlobalParameters() const {
+  return m_presenter.getGlobalParameters();
+}
+
+QStringList MSDTemplateBrowser::getLocalParameters() const {
+  return m_presenter.getLocalParameters();
+}
+
+void MSDTemplateBrowser::setGlobalParameters(const QStringList &globals) {
+  m_presenter.setGlobalParameters(globals);
+}
+
+void MSDTemplateBrowser::enumChanged(QtProperty *prop) {
+  if (!m_emitEnumChange)
+    return;
+  if (prop == m_fitType) {
+    auto fitType = m_enumManager->enumNames(prop)[m_enumManager->value(prop)];
+    m_presenter.setFitType(fitType);
+  }
+}
+
+void MSDTemplateBrowser::globalChanged(QtProperty *, const QString &, bool) {
+  //  std::cerr << "Global " << name.toStdString() << ' ' << on << std::endl;
```

Comment:
This method could be dropped since it does nothing

Reply:
Added comment to code